src/SfxWeb/src/app/modules/event-store/event-store/event-store.component.ts

```diff
@@ -58,92 +67,136 @@ export class EventStoreComponent implements OnInit, OnDestroy {
   public endDate: Date;
 
   ngOnInit() {
-    this.pshowAllEvents = !this.timelineGenerator;
-    this.resetSelectionProperties();
-    this.setTimelineData();
-    this.debouncerHandlerSubscription = this.debounceHandler
-    .pipe(debounceTime(400), distinctUntilChanged())
-    .subscribe(dates => {
-        this.startDate = dates.startDate;
-        this.endDate = dates.endDate;
-        this.setNewDateWindow();
-     });
+      this.pshowAllEvents = this.checkAllOption();
+      this.resetSelectionProperties();
+      this.setTimelineData();
+      this.debouncerHandlerSubscription = this.debounceHandler
+          .pipe(debounceTime(400), distinctUntilChanged())
+          .subscribe(dates => {
+              this.startDate = dates.startDate;
+              this.endDate = dates.endDate;
+              this.setNewDateWindow();
+          });
   }
 
   ngOnDestroy() {
       this.debouncerHandlerSubscription.unsubscribe();
   }
 
-  public reset(): void {
-      this.isResetEnabled = false;
-      if (this.eventsList.resetDateWindow()) {
-          this.resetSelectionProperties();
-          this.eventsList.reload().subscribe( data => {
-              this.setTimelineData();
-          });
-      } else {
-          this.resetSelectionProperties();
+  public checkAllOption(): boolean {
+      let onlyShowAll = false;
+      for (const data of this.listEventStoreData) {
+          if (!data.timelineGenerator) {
+              onlyShowAll = true;
+          }
       }
+      return onlyShowAll;
   }
 
   private resetSelectionProperties(): void {
-      this.startDate = this.eventsList.startDate;
-      this.endDate = this.eventsList.endDate;
+      this.startDate = this.listEventStoreData[0].eventsList.startDate;
+      this.endDate = this.listEventStoreData[0].eventsList.endDate;
       this.startDateMin = this.endDateMin = TimeUtils.AddDays(new Date(), -30);
       this.startDateMax = this.endDateMax = new Date(); // Today
   }
 
   public setDate(date: IQuickDates) {
       this.setNewDates({
-        endDate: new Date(this.eventsList.endDate),
-        startDate: TimeUtils.AddHours(this.endDate, -1 * date.hours)
+          endDate: new Date(this.listEventStoreData[0].eventsList.endDate),
+          startDate: TimeUtils.AddHours(this.endDate, -1 * date.hours)
       });
   }
 
   private setNewDateWindow(): void {
-      if (this.eventsList.setDateWindow(this.startDate, this.endDate)) {
-          this.resetSelectionProperties();
-          this.isResetEnabled = true;
-          this.eventsList.reload().subscribe( data => {
-              this.setTimelineData();
-          });
-      } else {
-          this.resetSelectionProperties();
+      let eventListChange = false;
+
+      for (const data of this.listEventStoreData) {
```

Comment:
given reload will require a http request, this will be non blocking below.
The code before would set the timeline data after it has waited for the reload function subscription to emit and then set the data. Like below, you should wait until all of them have reloaded and then set the data.

Reply:
I've added the forkJoin to wait until all observables have finished and also tried to include built in list functions, please let me know your feedback on this part.